cs/testing-with-travis.texy

```diff
@@ -0,0 +1,208 @@
+
+/--
+env:
+  - TESTER_PHP_BIN="php"
+  - TESTER_PHP_BIN="hhvm"
+\--
+
+Instalace závislostí
+====================
+
+.[caution]
+Návod předpokládá, že závislosti vašeho projektu jsou spravovány pomocí nástroje Composer a v kořenovém adresáři obsahuje `composer.json`. Zároveň musíte mít v závislostech uvedený nette/tester. Jak nastavit Composer najdete v kapitole [#Composer].
+
+V bloku `install` nainstalujeme pomocí Composeru závislosti našeho projektu. Každá odrážka znamená samostatný příkaz. Poté nainstalujeme závilosti včetně těch vývojářských `--dev`. Protože nechceme, aby se závislosti načítali z keše, použijeme `--prefer-source`. A nahonec, protože nechceme aby composer pokládal žádné interaktivní dotazy, na které by Travis nedokázal odpověděť, použijeme `--no-interaction`.
+
+Protože chceme používat aktuální verzi composeru, provedeme v bloku `before_install` jeho update.
+
+/--
+before_install:
+  - composer self-update
+
+install:
+  - composer install --no-interaction --prefer-source --dev
+\--
+
+Matice testů
+============
+
+Pro každý projekt je na základě předchozích údajů vytvořena testovací matice, tedy soubor testů, které budou spuštěny a obsahují veškeré možná kombinace. Pokud chceme některé kombinace vyřadit, můžeme využít blok `matrix`.
+Nebudeme chtít spouštět testy běžných verzí PHP s parametrem `-p hhvm`
+
+/--
+matrix:
+  exclude:
+    - php: 5.3
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.4
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.5
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.6
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: hhvm
+      env: TESTER_PHP_BIN="php"
+\--
+
+Aby testování bylo označeno Travisem jako úspěšné, musí projít všechny dílčí testy. Pokud nechceme, aby výsledek některé z nich ovlivňovaly, můžeme to deklarovat v bloku `allow_failures`. V našem případě nám nebude vadit, pokud neprojdou testy na HHVM.
+
+/--
+matrix:
+  allow_failures:
+    - php: hhvm
+\--
+
+Pokud nyní testy pod HHVM selžou, ale pod jinými verzemi projdou, bude build označen jako procházející.
+
+Spouštění testů
+===============
+
+Testy se spouštějí v sekci `script`, kde stačí zavolat tester. Předpokládejeme, že testy jsou umístěny ve složce `tests/` a používáte vlastní `php.ini` umístěné ve složce s testy.
+
+/--
+script:
+  - ./vendor/bin/tester -p $TESTER_PHP_BIN -c ./tests/php.ini ./tests
+\--
+
+Když se něco pokazí
+===================
+
+Travis nám umožňuje nastavit i blok `after_failure`, který se spustí v případě, že test selže. Pomocí tohoto bloku si můžeme například nechat vypsat výstupy funkcí, které se lišily od očekávaných výstupů uvedených v testech.
+
+/--
+after_failure:
+    # Vytiskne obsah souborů *.actual
+    - for i in $(find tests -name \*.actual); do echo "--- $i"; cat $i; echo; echo; done
```

Comment:
Možná radši `./tests` aby bylo jasné, že je to složka. Všude jinde to taky máš s tečkou na začátku.